File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let answer = 0;
    while (true) {
        let arr_x = arr.map((v) => {
            if (v % 2 === 0 && v >= 50) return v / 2
            else if (v % 2 !== 0 && v < 50) return v * 2 + 1
            else return v
        })
        if (arr.every((v,i) => v == arr_x[i])) return answer
        else {
            answer++
            arr = arr_x
        }
    }
    return answer;
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = 0
    while True:
        arr_x = []
        for v in arr:
            if v >= 50 and v % 2 == 0: arr_x.append(v/2)
            elif v < 50 and v % 2 != 0: arr_x.append(v*2 + 1)
            else: arr_x.append(v)
                
        if arr == arr_x: return answer
        else:
            answer += 1
            arr = arr_x